src/test/java/com/google/devtools/build/lib/skylark/SkylarkDefinedAspectsTest.java

```diff
@@ -904,6 +904,39 @@ public void aspectFailingOrphanArtifacts() throws Exception {
             + "test/missing_in_action.txt\n");
   }
 
+  @Test
+  public void aspectSkippingOrphanArtifactsWithLocation() throws Exception {
+    scratch.file("aspects/BUILD", "");
+    scratch.file(
+        "aspects/print.bzl",
+        "def _print_expanded_copts_impl(target, ctx):",
+        "    # Make sure the rule has a srcs attribute.",
+        "    if hasattr(ctx.rule.attr, 'cmd'):",
+        "        cmd = ctx.rule.attr.cmd",
+        "        print(\"{} -> {}\".format(cmd, ctx.expand_location(cmd, [])))",
```

Comment:
nit: just `ctx.expand_location(cmd, [])`? If you want to verify the result of the expansion, you can put it in a struct like e.g. line 2656 in `aspectOnAspectAttrConflict` (`return struct(result=<string>)`) then verify it like in line 2673 (`target.get("result")`)

Reply:
Good idea. Done.